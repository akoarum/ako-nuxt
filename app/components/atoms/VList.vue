<script>
export default {
  props: {
    tag: { type: String, default: 'ul' },
    visual: { type: String, default: 'bullet' },
    items: { type: Array, required: true }
  },
  // eslint-disable-next-line no-unused-vars
  render(h) {
    const Tag = this.tag
    const visualClass = `-${this.visual}`
    const items = this.items.map(item => <li>{item}</li>)

    return <Tag class={['list', visualClass]}>{items}</Tag>
  }
}
</script>

<style lang="scss" scoped>
.list {
  font-size: $fontSize-base;

  > li {
    position: relative;

    &::before {
      position: absolute;
    }

    + li {
      margin-top: 5px;
    }
  }

  &.-bullet {
    > li {
      padding-left: 16px;

      &::before {
        left: 4px;
        top: 0.6em;
        content: '';
        border-radius: 50%;
        width: 4px;
        height: 4px;
        background: currentColor;
      }
    }
  }
  &.-number {
    counter-reset: number;

    > li {
      padding-left: 28px;

      &::before {
        left: 0;
        counter-increment: number;
        content: counter(number) '.';
      }
    }
  }
}
</style>
