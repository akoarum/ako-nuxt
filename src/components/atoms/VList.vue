<script>
export default {
  props: {
    items: { type: Array, required: true },
    tag: { type: String, default: 'ul' },
    ordered: { type: Boolean }
  },
  computed: {
    styledClass() {
      if (this.ordered) return '-ordered'
      return ''
    }
  },
  render(h) {
    const Tag = this.tag
    const classes = ['vList'].concat([this.styledClass])
    return (
      <Tag class={classes.join(' ')}>
        {this.items.map(item => (
          <li class="vList__item" key={item.id}>
            {item.display}
          </li>
        ))}
      </Tag>
    )
  }
}
</script>

<style lang="scss" scoped>
.vList {
  @include media-min {
    font-size: $fontSize-pc;
  }

  @include media-max {
    font-size: $fontSize-sp;
  }
}

.vList__item {
  position: relative;

  &::before {
    position: absolute;
    left: 2px;
    content: '';
    border-radius: 50%;
    width: 5px;
    height: 5px;
    background: $color-theme;
  }

  @include media-min {
    padding-left: 16px;

    &::before {
      top: 0.55em;
    }
  }

  @include media-max {
    padding-left: 14px;

    &::before {
      top: 0.6em;
    }
  }
}

.vList__item + .vList__item {
  @include media-min {
    margin-top: 4px;
  }

  @include media-max {
    margin-top: 3px;
  }
}

.-ordered .vList {
  counter-reset: count;
}

.-ordered .vList__item {
  padding-left: 1.4em;
  counter-increment: count;

  &::before {
    top: 0.05em;
    left: 0;
    content: counter(count) '.';
    border-radius: 0;
    width: auto;
    height: auto;
    background: transparent;
  }
}
</style>
