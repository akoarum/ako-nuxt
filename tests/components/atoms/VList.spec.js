import { shallowMount } from '@vue/test-utils'
import VList from '~/components/atoms/VList'

describe('VList', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VList, {
      propsData: {
        items: [
          { id: 1, display: 'テスト' },
          { id: 2, display: 'テスト2' }
        ]
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClass: propsを元にクラスを生成して返す', () => {
      expect(VList.computed.styledClass.call({ ordered: true })).toBe('-ordered')
      expect(VList.computed.styledClass.call({})).toBe('')
    })
  })
})
