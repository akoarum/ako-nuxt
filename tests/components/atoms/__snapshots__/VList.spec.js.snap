// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VList initialized snapshot 1`] = `
<ul
  class="vList "
>
  <li
    class="vList__item"
  >
    テスト
  </li>
  <li
    class="vList__item"
  >
    テスト2
  </li>
</ul>
`;
